common/djangoapps/microsite_configuration/microsite.py

```diff
@@ -98,13 +121,23 @@ def get_value_for_org(org, val_name, default=None):
 
 def get_all_orgs():
     """
-    This returns a set of orgs that are considered within a microsite. This can be used,
+    This returns a set of orgs that are considered within all microsites. This can be used,
     for example, to do filtering
     """
     org_filter_set = set()
     if not has_configuration_set():
         return org_filter_set
 
+    if "microsite_configuration.middleware.DatabaseMicrositeMiddleware" in settings.MIDDLEWARE_CLASSES:
+        # Get the orgs in the db
+        candidates = Microsite.objects.all()
+        for microsite in candidates:
+            current = json.loads(microsite.values)
+            org_filter = current.get('course_org_filter')
+            if org_filter:
+                org_filter_set.add(org_filter)
+
+    # Get the orgs in the settings file
```

Comment:
I would prefer setting up an interface layer here rather than doing conditional logic. Seems like we need a MicrositeConfigurationBackend abstract interface with two implementations of the interface: MicrositeDjangoSettingsConfigurationBackend and MicrositeDatabaseConfigurationBackend.

What do you think about that? I don't think the interface would need a lot of methods.